File: android/src/main/java/com/quito/models/PublishOptions.kt
package com.quito.models

import com.facebook.react.bridge.ReadableMap
import com.quito.utils.getOr

data class PublishOptions(
  val qos: QoS,
  val retain: Boolean,
  val isDuplicate: Boolean,
) {
  constructor(pubOptionsFromJs: ReadableMap): this(
    QoS.values()[pubOptionsFromJs.getOr<Int>("qos", 0)],
    pubOptionsFromJs.getOr("retain", false),
    pubOptionsFromJs.getOr("isDuplicate", false)
  )
}

File: ios/PublishOptions.swift
struct PublishOptions {
  let qos: QoS
  let retain: Bool
  let isDuplicate: Bool

  init(fromJsPubOptions pubOptionsFromJs: NSDictionary) {
      self.qos = QoS(rawValue: Helpers.getOrDefault(dict: pubOptionsFromJs, key: "qos", defaultValue: 0))!
      self.retain = Helpers.getOrDefault(dict: pubOptionsFromJs, key: "retain", defaultValue: false)
      self.isDuplicate = Helpers.getOrDefault(dict: pubOptionsFromJs, key: "isDuplicate", defaultValue: false)
  }
}
